apps/governance/src/components/telemetry-dialog/telemetry-dialog.tsx

```diff
@@ -0,0 +1,129 @@
+import { create } from 'zustand';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useLocalStorage } from '@vegaprotocol/react-helpers';
+import { Dialog, Icon, Button } from '@vegaprotocol/ui-toolkit';
+import { Networks, useEnvironment } from '@vegaprotocol/environment';
+
+type TelemetryDialogState = {
+  isOpen: boolean;
+  open: () => void;
+  close: () => void;
+};
+
+const useTelemetryDialogStore = create<TelemetryDialogState>((set) => ({
+  isOpen: false,
+  open: () => set({ isOpen: true }),
+  close: () => set({ isOpen: false }),
+}));
+
+export const TELEMETRY_DIALOG_PREVIOUSLY_OPENED =
+  'telemetry_dialog_previously_opened';
+export const TELEMETRY_ON = 'telemetry_on';
+
+export const useTelemetryDialog = () => {
+  const [getPreviouslyOpened, setPreviouslyOpened] = useLocalStorage(
+    TELEMETRY_DIALOG_PREVIOUSLY_OPENED
+  );
+  const [getTelemetryOn, setTelemetryOn] = useLocalStorage(TELEMETRY_ON);
+
+  const { VEGA_ENV } = useEnvironment();
+  const isMainnet = VEGA_ENV === Networks.MAINNET;
+
+  const defaultTelemetryAccepted = isMainnet ? 'false' : 'true';
+
+  const { isOpen, open, close } = useTelemetryDialogStore();
+
+  const [telemetryAccepted, setTelemetryAcceptedState] = useState(
+    JSON.parse(getTelemetryOn || defaultTelemetryAccepted)
+  );
+
+  useEffect(() => {
```

Comment:
suggestion (non blocking): find it quite confusing having two useLocalStorage instances and two useEffects

can we tidy this up a bit, something along the lines of the below should work nicely
```

const useTelemetryDialogStore = create<TelemetryDialogState>((set) => ({
  isOpen: LocalStorage.getItem(TELEMETRY_ON) === null ? true : false,
  open: () => set({ isOpen: true }),
  close: () => set({ isOpen: false }),
}));

export const useTelemetryDialog = () => {
  const [telemetryAccepted, setTelemetryOn] = useLocalStorage(TELEMETRY_ON);
  const { isOpen, open, close } = useTelemetryDialogStore();

  return {
    isOpen,
    open,
    close,
    telemetryAccepted,
    setTelemetryAccepted: (value: boolean) => {
      setTelemetryOn(value.toString());
    },
  };
};
```